app/scripts/migrations/120.5.ts

```diff
@@ -0,0 +1,126 @@
+ *
+ * Hard-coded here rather than imported so that this migration continues to work correctly as these
+ * constants get updated in the future.
+ */
+const infuraNetworkTypes = [
+  'mainnet',
+  'goerli',
+  'sepolia',
+  'linea-goerli',
+  'linea-sepolia',
+  'linea-mainnet',
+];
+
+/**
+ * Remove invalid network configuration IDs from the SelectedNetworkController.
+ *
+ * @param state - The persisted MetaMask state, keyed by controller.
+ */
+function transformState(state: Record<string, unknown>): void {
+  if (!hasProperty(state, 'SelectedNetworkController')) {
+    return;
+  }
+  if (!isObject(state.SelectedNetworkController)) {
+    console.error(
+      `Migration ${version}: Invalid SelectedNetworkController state of type '${typeof state.SelectedNetworkController}'`,
+    );
+    delete state.SelectedNetworkController;
+    return;
+  } else if (!hasProperty(state.SelectedNetworkController, 'domains')) {
+    console.error(
+      `Migration ${version}: Missing SelectedNetworkController domains state`,
+    );
+    delete state.SelectedNetworkController;
+    return;
+  } else if (!isObject(state.SelectedNetworkController.domains)) {
+    console.error(
+      `Migration ${version}: Invalid SelectedNetworkController domains state of type '${typeof state
+        .SelectedNetworkController.domains}'`,
+    );
+    delete state.SelectedNetworkController;
+    return;
+  }
+
+  if (!hasProperty(state, 'NetworkController')) {
+    delete state.SelectedNetworkController;
+    return;
+  } else if (!isObject(state.NetworkController)) {
+    console.error(
+      new Error(
+        `Migration ${version}: Invalid NetworkController state of type '${typeof state.NetworkController}'`,
+      ),
+    );
+    delete state.SelectedNetworkController;
+    return;
+  } else if (!hasProperty(state.NetworkController, 'networkConfigurations')) {
+    delete state.SelectedNetworkController;
+    return;
+  } else if (!isObject(state.NetworkController.networkConfigurations)) {
+    console.error(
+      new Error(
+        `Migration ${version}: Invalid NetworkController networkConfigurations state of type '${typeof state.NetworkController}'`,
+      ),
+    );
+    delete state.SelectedNetworkController;
+    return;
+  }
+
+  const validNetworkConfigurationIds = [
+    ...infuraNetworkTypes,
+    ...Object.keys(state.NetworkController.networkConfigurations),
+  ];
+  const domainMappedNetworkConfigurationIds = Object.values(
+    state.SelectedNetworkController.domains,
+  );
+
+  for (const configurationId of domainMappedNetworkConfigurationIds) {
+    if (
+      typeof configurationId !== 'string' ||
+      !validNetworkConfigurationIds.includes(configurationId)
+    ) {
+      console.error(
+        new Error(
+          `Migration ${version}: Invalid networkConfigurationId found in SelectedNetworkController state: '${configurationId}'`,
+        ),
+      );
+      delete state.SelectedNetworkController;
```

Comment:
thoughts on deleting just the invalid domain entry?

Reply:
We definitely could do that. I figured that this would be simpler, and we already chose this strategy in v12.0.1 for handling invalid state (rather than fixing it properly). The impact of deleting this state is rather minimal. Slight annoyance at worst.

Reply:
fair. definitely a nit